src/activity_item_handling.cpp

```diff
@@ -1200,64 +1244,113 @@ static activity_reason_info can_do_activity_there( const activity_id &act, playe
-                std::vector<tripoint> points_to_check;
-                for( const auto elem : g->m.points_in_radius( src_loc, PICKUP_RANGE - 1 ) ) {
-                    points_to_check.push_back( elem );
-                }
-                const inventory &inv = p.crafting_inventory();
-                const bool can_make = reqs.can_make_with_inventory( inv, is_crafting_component );
-                p.set_value( "veh_index_type", vpinfo.name() );
-                // temporarily store the intended index, we do this so two NPCs dont try and work on the same part at same time.
-                p.activity_vehicle_part_index = vpindex;
-                if( !can_make ) {
-                    return activity_reason_info::fail( NEEDS_VEH_DECONST );
-                } else {
-                    return activity_reason_info::ok( NEEDS_VEH_DECONST );
+            } else if( act == activity_id( "ACT_VEHICLE_REPAIR" ) ) {
+                // find out if there is a vehicle part here we can repair.
+                std::vector<vehicle_part *> parts = veh->get_parts_at( src_loc, "", part_status_flag::any );
+                for( vehicle_part *part_elem : parts ) {
+                    const vpart_info &vpinfo = part_elem->info();
+                    int vpindex = veh->index_of_part( part_elem, true );
+                    // if part is undamaged or beyond repair - can skip it.
+                    if( part_elem->is_broken() || part_elem->damage() == 0 ) {
+                        continue;
+                    }
+                    for( const npc &guy : g->all_npcs() ) {
+                        if( guy.disp_name() != p.disp_name() && guy.activity_vehicle_part_index != -1 &&
+                            guy.activity_vehicle_part_index == vpindex ) {
+                            continue;
+                        }
+                    }
+                    // if the vehicle is moving or player is controlling it.
+                    if( abs( veh->velocity ) > 100 || veh->engine_on ) {
+                        continue;
+                    }
+                    // dont have skill to remove it
+                    std::map<skill_id, int> repair_skills = vpinfo.repair_skills;
+                    for( const auto &e : repair_skills ) {
+                        bool hasSkill = p.get_skill_level( e.first ) >= e.second;
+                        if( !hasSkill ) {
+                            continue;
+                        }
+                    }
+                    const auto &reqs = vpinfo.repair_requirements();
+                    const std::string ran_str = random_string( 10 );
+                    const requirement_id req_id( ran_str );
+                    requirement_data::save_requirement( reqs, req_id );
+                    std::vector<tripoint> points_to_check;
+                    for( const auto elem : g->m.points_in_radius( src_loc, PICKUP_RANGE - 1 ) ) {
+                        points_to_check.push_back( elem );
```

Comment:
This is even more nonsensical. The variable `points_to_check` is set, but never read.